### ISSUE TITLE
Beacon 2.2.0 crashes when using Kotlin 1.4

### ISSUE THREAD

**Issue by samruston** [helpscout/beacon-android-sdk-sample#124]
Thank you for taking the time to submit an issue with all the details shown below. Our engineering team monitors issues submitted and strives to respond with 1-2 business days.

* Beacon version: 2.2.0
* Android OS version: 10
* Device make/model: Emulator & Google Pixel

**Steps to reproduce**
1. Set dependencies to Kotlin 1.4 & Beacon 2.2.0 in android sample
2. Try pressing "Open beacon"

```
java.lang.NoSuchMethodError: No virtual method elapsedNow()D in class Lkotlin/time/TimeMark; or its super classes (declaration of 'kotlin.time.TimeMark' appears in /data/app/~~Q7YN7au5Osfn5wBEJjTiKA==/net.helpscout.samples.beacon.kotlin-mzPaV7GcyG5C4nYxY5eCGg==/base.apk!classes3.dex)
        at org.koin.core.time.MeasureKt.measureDuration(Measure.kt:36)
        at org.koin.core.KoinApplication.modules(KoinApplication.kt:60)
        at com.helpscout.beacon.a.d.a.c$a$a.a(Unknown Source:17)
        at com.helpscout.beacon.a.d.a.c$a$a.invoke(Unknown Source:2)
        at org.koin.dsl.KoinApplicationKt.koinApplication(KoinApplication.kt:28)
        at com.helpscout.beacon.a.d.a.c$a.a(Unknown Source:22)
        at com.helpscout.beacon.a.d.a.c$a.a(Unknown Source:6)
        at com.helpscout.beacon.internal.presentation.common.c.s(Unknown Source:13)
        at com.helpscout.beacon.internal.presentation.common.c.onCreate(Unknown Source:3)
        at com.helpscout.beacon.internal.presentation.ui.home.HomeActivity.onCreate(Unknown Source:0)
        at android.app.Activity.performCreate(Activity.java:8000)
        at android.app.Activity.performCreate(Activity.java:7984)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**Comment by aliripe** [comment]
Hi Sam,

Thank you so much for reporting this! It looks like Koin – one of the libraries that we're using internally – is presenting issues with Kotlin 1.4. They're looking into it https://github.com/InsertKoinIO/koin/pull/873 and have released an alpha version with a fix. 

We will, however, hold off updating the Beacon SDK library for the time being until they release a stable version so my suggestion would be that you temporarily revert to Kotlin 1.3.72 for the time being until this issue gets resolved. 

I have created a ticket internally to keep track of this and we will make sure to let you know once a fix is out!

Ticket: [ANDSDK-568](https://helpscout.atlassian.net/browse/ANDSDK-568)

**Comment by scottyab** [comment]
Just checking in here. There's a Koin `2.2.0-beta-1` out and we'll be looking to update the Beacon SDK dependancy once this is marked stable. Thanks

**Comment by aliripe** [comment]
@samruston Hi there! Alicia, hopping in with Scott, 

While we wait for the latest stable version to be released, if you want to use Kotlin 1.4 you should be able to do so by forcing our SDK to use the latest version of Koin. You can do so adding the following piece of code to your root `build.gradle` file:

```
allprojects {
    ...

    ext.koinVersion = '2.2.0-beta-1'

    configurations.all {
        resolutionStrategy {
            force "org.koin:koin-core:$koinVersion"
            force "org.koin:koin-android:$koinVersion"
            force "org.koin:koin-androidx-scope:$koinVersion"
            force "org.koin:koin-androidx-viewmodel:$koinVersion"
        }
    }
}
```

Let us know if it works!

**Comment by samruston** [comment]
@aliripe Unfortunately this doesn't seem to fix it.

Though it now crashes with a different error (reproduced in sample app):

```
java.lang.NoSuchMethodError: No virtual method getRootScope()Lorg/koin/core/scope/ScopeDefinition; in class Lorg/koin/core/module/Module; or its super classes (declaration of 'org.koin.core.module.Module' appears in /data/app/~~ZacQ37NMPbLpZ4L_7HWGgQ==/net.helpscout.samples.beacon.kotlin-9rCz-OBei7KksHUGTKjpIQ==/base.apk!classes3.dex)
        at com.helpscout.beacon.internal.presentation.inject.modules.AppModuleKt$appModule$1.invoke(Unknown Source:9)
        at com.helpscout.beacon.internal.presentation.inject.modules.AppModuleKt$appModule$1.invoke(Unknown Source:2)
        at org.koin.dsl.ModuleKt.module(Module.kt:31)
        at org.koin.dsl.ModuleKt.module$default(Module.kt:29)
        at com.helpscout.beacon.internal.presentation.inject.modules.AppModuleKt.<clinit>(Unknown Source:6)
        at com.helpscout.beacon.internal.presentation.inject.modules.AppModuleKt.getAppModule(Unknown Source:0)
        at com.helpscout.beacon.a.d.a.c$a.<clinit>(Unknown Source:18)
        at com.helpscout.beacon.a.d.a.c.<clinit>(Unknown Source:0)
        at com.helpscout.beacon.internal.presentation.common.c.s(Unknown Source:0)
        at com.helpscout.beacon.internal.presentation.common.c.onCreate(Unknown Source:3)
        at com.helpscout.beacon.internal.presentation.ui.home.HomeActivity.onCreate(Unknown Source:0)
        at android.app.Activity.performCreate(Activity.java:8000)
        at android.app.Activity.performCreate(Activity.java:7984)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**Comment by sipersso** [comment]
This issue has apparently been open for 3 months, so how knows when there will be a new release: 
https://github.com/InsertKoinIO/koin/issues/847

@scottyab and the helpscout team. If you don't want to wait for the next Koin releaes this can apparently be fixed by changing the Koin log level. Would this be something to try in the meantime? 
```
startKoin {
        androidLogger(Level.NONE)          
 }
```

https://stackoverflow.com/questions/63393507/java-lang-nosuchmethoderror-no-virtual-method-elapsednowd-in-class-lkotlin-ti

**Comment by aliripe** [comment]
@sipersso Great suggestion, again, thank you so much for helping out!

@samruston Since this issue and #132 are caused by problems with Koin, let's move the discussion to #132 so that no information gets lost for all parties involved.

**Comment by aliripe** [comment]
Hi @samruston @sipersso, I'm pleased to mention we have fixed this issue as part of our 2.2.2 release. Please see the [changelog](https://github.com/helpscout/beacon-android-sdk-sample/blob/master/CHANGELOG.md) for full details.

**Comment by samruston** [comment]
@aliripe Awesome, thanks
